src/insightsAPIBeaconClient.test.ts

```diff
@@ -48,6 +48,44 @@ describe('InsightsApiBeaconClient', () => {
     expect(fetch).toHaveBeenCalledTimes(1);
   });
 
+  test('it uses credentials from the constructor by default', () => {
+    const beacon = new InsightsApiBeaconClient(clientOpts);
+    beacon.send(testEvent);
+
+    expect(fetch).toHaveBeenCalledWith(
+      expect.any(String),
+      expect.objectContaining({
+        headers: expect.objectContaining({
+          'X-Algolia-Application-Id': clientOpts.applicationId,
+          'X-Algolia-API-Key': clientOpts.apiKey,
+        }),
+      })
+    );
+  });
+
+  test('it overrides credentials if found in the event payload', () => {
```

Comment:
Can we also test that credentials are removed from the payload, and aren't forwarded in the event body to the Insights API?

Reply:
Done in https://github.com/algolia/search-insights.js/pull/398/commits/a58248ff3fcbee5187a4d16254714523088d9d0f